### ISSUE TITLE
data global undefined when using py renderer

### ISSUE THREAD

**Issue by kylehuff** [saltstack/salt#50824]
### Description of Issue/Question
The [docs](https://docs.saltstack.com/en/latest/ref/renderers/all/salt.renderers.py.html) indicate the `data` object is available in a pure python state (py renderer), the same as in a jinja template -- but that does not appear to be the case. Other global variables appear to be present.

### Setup
```
$ cat /srv/salt/states/python_state/init.sls 
#!py

def run():
  config = {}

  # showing dunders are available
  sls = __sls__
  salt = __salt__
  env = __env__

  # data not available (NameError exception)
  my_data = data

  return config
```

### Steps to Reproduce Issue
```
$ salt-call state.apply python_state
[CRITICAL] Rendering SLS 'base:python_state' failed: Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/salt/utils/templates.py", line 566, in py
    data = mod.run()
  File "/var/cache/salt/minion/files/base/python_state/init.sls", line 10, in run
    my_data = data
NameError: global name 'data' is not defined

local:
    Data failed to compile:
----------
    Rendering SLS 'base:python_state' failed: Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/salt/utils/templates.py", line 566, in py
    data = mod.run()
  File "/var/cache/salt/minion/files/base/python_state/init.sls", line 10, in run
    my_data = data
NameError: global name 'data' is not defined
```

### Versions Report
(local salt-minion running on salt-master, pointed at localhost, both minion+master v2018.3.3)
```
Salt Version:
           Salt: 2018.3.3
 
Dependency Versions:
           cffi: 1.11.5
       cherrypy: Not Installed
       dateutil: 2.5.3
      docker-py: Not Installed
          gitdb: 2.0.0
      gitpython: 2.1.1
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.6
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: 2.18
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.13 (default, Nov 24 2017, 17:33:09)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 16.0.2
           RAET: Not Installed
          smmap: 2.0.1
        timelib: Not Installed
        Tornado: 4.4.3
            ZMQ: 4.2.1
 
System Versions:
           dist: debian 9.4 
         locale: UTF-8
        machine: x86_64
        release: 4.9.0-6-amd64
         system: Linux
        version: debian 9.4 
```

**Comment by kylehuff** [comment]
> What are you trying to achieve by using data? Maybe i can help point you to an alternative context.

TLDR;
I don't actually need that functionality, I just ran into it when checking something.

I merely wanted access to a variable defined within a `jinja|yaml` SLS that was using `slsutil.renderer` to render a pure-python state.  This is trivial by simply passing it as an argument to the `slsutil.renderer` function, but I first wanted to make sure it wasn't already being exposed via some global -- which is when I discovered that the docs did not seem to match the behavior of the global `data` in the py renderer in this context.

Thanks for the response and subsequent fix.

**Comment by darkpixel** [comment]
Go away stalebot.

**Comment by sagetherage** [comment]
@saltstack/docs-working-group

**Comment by kylehuff** [comment]
This appears to be 'not a bug' but rather misleading documentation that has been corrected and merged; closing.
